tls/s2n_config.c

```diff
@@ -808,6 +815,37 @@ int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_t
     return 0;
 }
 
+int s2n_config_add_custom_x509_extension(struct s2n_config *config, uint8_t *extension_oid, uint32_t extension_oid_len)
+{
+    POSIX_ENSURE(config, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(extension_oid, S2N_ERR_INVALID_ARGUMENT);
+
+    POSIX_ENSURE(s2n_libcrypto_supports_custom_oid(), S2N_ERR_API_UNSUPPORTED_BY_LIBCRYPTO);
+
+    if (config->custom_x509_extension_oids == NULL) {
+        config->custom_x509_extension_oids = sk_ASN1_OBJECT_new_null();
+    }
+    POSIX_ENSURE_REF(config->custom_x509_extension_oids);
+
+    DEFER_CLEANUP(struct s2n_stuffer oid_buffer = { 0 }, s2n_stuffer_free);
+    uint32_t buf_size = extension_oid_len + 1;
+    POSIX_GUARD(s2n_stuffer_alloc(&oid_buffer, buf_size));
+
+    uint8_t null_terminator = '\0';
+    POSIX_GUARD(s2n_stuffer_write_bytes(&oid_buffer, extension_oid, extension_oid_len));
+    POSIX_GUARD(s2n_stuffer_write_bytes(&oid_buffer, &null_terminator, 1));
+    POSIX_ENSURE_EQ(s2n_stuffer_data_available(&oid_buffer), buf_size);
+
+    const char *oid_string = s2n_stuffer_raw_read(&oid_buffer, buf_size);
+    POSIX_ENSURE_REF(oid_string);
```

Comment:
Nit: This might actually be simpler if you don't use stuffers. Do you think this version would be better, worse, or the same?

```
DEFER_CLEANUP(struct s2n_blob oid_buffer = { 0 }, s2n_free);
POSIX_GUARD(s2n_alloc(&oid_buffer, extension_oid_len + 1);
POSIX_GUARD(s2n_blob_zero(&oid_buffer);
POSIX_CHECKED_MEMCPY(oid_buffer.data, extension_oid, extension_oid_len);
const char *oid_string = oid_buffer.data;
```
But maybe '\0' == 0 isn't obvious? you could also do `oid_buffer.data[extension_oid_len] = '\0'` rather than just relying on s2n_blob_zero.

Reply:
> But maybe '\0' == 0 isn't obvious?

Also the + 1 to `extension_oid_len` is pretty important in this example. Having `oid_buffer.data[extension_oid_len] = '\0'` might also be preferable to make this explicit.